packages/lib/gpc/src/gpcUtil.ts

```diff
@@ -405,71 +524,88 @@ export function GPCRequirements(
+    const pod = proofConfig.pods[podName];
+
+    for (const entryName of Object.keys(pod.entries)) {
+      const lists = pod.entries[entryName].isMemberOf;
+
+      if (lists !== undefined) {
+        if (Array.isArray(lists)) {
+          if (lists.length === 0) {
             throw new TypeError(
               `The list of lists of valid values for ${podName}.${entryName} is empty.`
             );
           }
-          return lists === undefined
-            ? []
-            : lists.map((listName): [PODName, PODEntryIdentifier] => [
-                listName,
-                `${podName}.${entryName}`
-              ]);
-        })
-        .flat();
-      return listEntryPairs;
-    })
-    .flat();
+          for (const listName of lists ?? []) {
+            entryLists.push([listName, `${podName}.${entryName}`]);
+          }
+        } else {
+          entryLists.push([lists, `${podName}.${entryName}`]);
+        }
+      }
+    }
+  }
 
   // Find all [listName, tupleID] pairs in proofConfig.tuples
-  const tupleLists: [PODName, TupleIdentifier][] = Object.keys(
-    proofConfig.tuples ?? {}
-  )
-    .map((tupleName): [PODName, TupleIdentifier][] => {
-      const lists = (proofConfig.tuples ?? {})[tupleName].liesInLists;
-      if (lists !== undefined && lists.length === 0) {
-        throw new TypeError(
-          `The list of lists of valid values for tuple.${JSON.stringify(
-            tupleName
-          )} is empty.`
-        );
+  const tupleLists: [PODName, PODEntryIdentifier][] = [];
+
+  for (const tupleName of Object.keys(proofConfig.tuples ?? {})) {
+    const lists = (proofConfig.tuples ?? {})[tupleName].isMemberOf;
+
+    if (lists !== undefined) {
+      if (Array.isArray(lists)) {
+        if (lists.length === 0) {
+          throw new TypeError(
+            `The list of lists of valid values for $tuple.${tupleName} is empty.`
+          );
+        }
+        for (const listName of lists ?? []) {
+          entryLists.push([listName, `$tuple.${tupleName}`]);
+        }
+      } else {
+        entryLists.push([lists, `$tuple.${tupleName}`]);
       }
-      return lists === undefined
-        ? []
-        : lists.map((listName) => [listName, `tuple.${tupleName}`]);
-    })
-    .flat();
+    }
+  }
 
   // Combine the two and compile config.
-  return entryLists
-    .concat(tupleLists)
-    .reduce(
-      (
-        listConfig: GPCProofMembershipListConfig,
-        listIdPair: [PODName, PODEntryIdentifier]
-      ): GPCProofMembershipListConfig => {
-        const [listName, identifier] = listIdPair;
-        const currentPair = listConfig[listName] ?? [];
-        return {
-          ...listConfig,
-          [listName]: currentPair.concat([identifier])
-        };
-      },
-      {}
+  const listConfig: GPCProofMembershipListConfig = {};
+
+  for (const [listName, id] of entryLists.concat(tupleLists)) {
+    if (listConfig[listName] === undefined) {
```

Comment:
Good modification of my suggestion version.  I think the if/else is slightly clearer than my first attempt at handling the first-entry special case.